recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
+        return self._parasite
+
+    @property
+    def host(self) -> str:
+        return self._host
+    
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self._parasite == other.parasite and self._host == other.host
+        return False
+    
+    def __hash__(self):
+        return hash((self._parasite, self._host))
+    
+    def __repr__(self):
+        return "%s(%s, %s)" % (type(self).__name__, self._parasite, self._host)
```

Comment:
See what I said before about type.  Do you want type(self).__name__ to return the class of self, even if it is inherited, or do you want it to say "MappingNode"?

Reply:
I see. I think it makes sense for it to display MappingNode if it is inherited and the superclass does not define its own repr.